compiler/rustc_builtin_macros/src/format_foreign.rs

```diff
@@ -13,10 +13,10 @@ pub(crate) mod printf {
     }
 
     impl<'a> Substitution<'a> {
-        pub(crate) fn as_str(&self) -> &str {
+        pub(crate) fn as_string(&self) -> String {
```

Comment:
probably call this [`to_string`](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv), or maybe just implement `ToString`